api/views.py

```diff
@@ -49,6 +49,7 @@ def get_queryset(self):
 class ProfileViewSet(viewsets.ModelViewSet):
     serializer_class = ProfileSerializer
     permission_classes = [permissions.IsAuthenticated, IsOwner]
+    http_method_names = ['get', 'post', 'head', 'put', 'patch']
```

Comment:
```suggestion
    http_method_names = ['get', 'patch']
```
Can we consider restricting this more? Not sure what methods the add-on/client uses to update the `server_storage` and the `onboarding_state` but we could consider removing `post`, `patch`, and `put` if it only uses `patch` for example. Would prefer the `patch` over `put` since `put` historically was used to create object if it didn't exist (no longer the case after 3.0, see doc [here](https://www.django-rest-framework.org/api-guide/generic-views/#put-as-create))

Reply:
I'd like to leave the API endpoint more open-ended to accomodate more client use-cases. (I'm also thinking we should go back thru and expose MORE of the model in the endpoints, e.g., I had to [add `full_address` to the address API endpionts.](https://github.com/mozilla/fx-private-relay/pull/1425))